app/graphql/types/query_type.rb

```diff
@@ -21,11 +24,12 @@ def nodes(ids:)
     field :get_referral, Types::ReferralType, null: false
 
     def get_referral
-      context.authenticate_user!
-      # TO-DO: Implement policies with Pundit
-      # to check if the current_user is allowed to
-      # access the referral token (and other resources)
-      referral_token = context.current_user.authenticatable.referral_token
+      authorize(current_user, :get_referral_token?)
```

Comment:
From now on we can use `authorize` with a custom policy to authorize and authenticate at the same time.

```rb
# authorize() will authenticate + authorize
authorize(object, action)
```

If we don't need to authorize but authenticate we could continue using `authenticate_user!`.

---

We no longer need to call these methods through the `context` object, because now we have the `Authorization` and
`Context` helpers included over the `GraphQL` layer.